polkadot/roadmap/implementers-guide/src/runtime/parainherent.md

```diff
@@ -60,3 +60,32 @@ processing it, so the processed inherent data is simply dropped.
 This also means that the `enter` function keeps data around for no good reason. This seems acceptable though as the size
 of a block is rather limited. Nevertheless if we ever wanted to optimize this we can easily implement an inherent
 collector that has two implementations, where one clones and stores the data and the other just passes it on.
+
+## Data sanitization
+`ParasInherent` performs sanitization on the provided input data. When the module is invoked via `create_inherent` the
+input data is sanitized. `enter` entry point requires sanitized input. If unsanitized data is provided the module
+generates an error.
```

Comment:
```suggestion
`ParasInherent` with the entry point of `create_inherent` sanitizes the input data, while the `enter` entry point enforces already sanitized input data. If unsanitized data is provided the module generates an error.
```
Only minor rewording as
> `ParasInherent` performs sanitization on the provided input data.

technically isn't correct as `enter` does not truly sanitize data instead as you point out it checks for sanitization. Minor difference but want to make it clear what is the intent. (I know that we check for sanitization by effectively re-performing it but that step could be done differently.)